frontend/components/site/SiteSettings/BasicAuthSettingsForm.jsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Field, reduxForm } from 'redux-form';
+import InputWithErrorField from '../../Fields/InputWithErrorField';
+import { validBasicAuthUsername, validBasicAuthPassword } from '../../../util/validators';
+import { BASIC_AUTH } from '../../../propTypes';
+
+export const validateBasicAuthPassword = (value) => {
+  if (value && value.length && !validBasicAuthPassword(value)) {
+    return 'Password may contain alphanumeric characters and symbols !@$. Minimum length 6 characters.';
+  }
+
+  return undefined;
+};
+
+export const validateBasicAuthUsername = (value) => {
+  if (value && value.length && !validBasicAuthUsername(value)) {
+    return 'Username is invalid. Only alphanumeric characters are allowed. Minimum length 4 characters.';
+  }
+
+  return undefined;
+};
+
+export const BasicAuthSettingsForm = ({
+  handleSubmit, invalid, pristine, submitting,
+}) => (
+  <form className="settings-form" onSubmit={data => handleSubmit(data)}>
+    <h3>Basic Authentication Settings</h3>
+    <div className="well">
+      <fieldset>
+        <p className="well-text">
+          To enable basic authentication, please submit a username and password credentials required to preview your site builds.
+        </p>
+        <Field
+          name="username"
+          type="text"
+          label="Username:"
+          component={InputWithErrorField}
+          required
+          validate={[validateBasicAuthUsername]}
+          placeholder="username"
```

Comment:
Maybe omit the placeholder, users can find them confusing and we already have a nice label